### PR TITLE
update regexp which cleaned root path

### LINKED ISSUE #12805: Nested scope in route removes slash at the beginning of the url
First example:

``` ruby
# config/routes.rb
scope "(:locale)" do
  root_to "welcome#index"
end
```

When i run `rake routes` result is:
`web/welcome#index      root      /(:locale)(.:format)`
and helper method returns
`root_path #=>"/"`

If i add second scope:

``` ruby
# config/routes.rb
scope "(:locale)" do
  scope "(:platform)" do
    root_to "welcome#index"
  end
end
```

Result is:
`web/welcome#index      root      (/:locale)(/:platform)(.:format)`
and
`root_path #=> ""` - empty string

The first slash in the url is inside the optional parameter. It is correct?

### PR DESCRIPTION
**Pull request by tarasmatsyk**
Closes #12805.
Slash was removed in case of nested scope due to mismatching unless statement.

### CODE REVIEW

**Comment by maclover7** [comment]
Can you add a regression test here?

**Comment by tarasmatsyk** [comment]
Sure, give me some time.

**Comment by tarasmatsyk** [comment]
Ok, I covered as much cases as I found. I also forgot to replace "/(/:locale)(/:platform)" with "foo/bar" or smth like that. 
Could you please check if I done everything correctly?

**Comment by tarasmatsyk** [comment]
@maclover7, Happy New Year, Jon!
Accidentally found this old MR and thought why not to update and try to merge it.  All tests have been fixed and the branch is up to date, would appreciate if it can be accepted.

File: actionpack/lib/action_dispatch/routing/mapper.rb [hunk 1]
```diff
@@ -387,10 +387,10 @@ def dispatcher(raise_on_name_error)
 
       # Invokes Journey::Router::Utils.normalize_path and ensure that
       # (:locale) becomes (/:locale) instead of /(:locale). Except
-      # for root cases, where the latter is the correct one.
+      # for root cases including nesting up to second level.
       def self.normalize_path(path)
         path = Journey::Router::Utils.normalize_path(path)
-        path.gsub!(%r{/(\(+)/?}, '\1/') unless path =~ %r{^/\(+[^)]+\)$}
+        path.gsub!(%r{/(\(+)/?}, '\1/') unless path =~ %r{^/(\(+[^)]+\)){1,2}$}
```
Comment (gmcgibbon) [inline actionpack/lib/action_dispatch/routing/mapper.rb]: I would revert the comment change and make the regex allow any number of nested optional scopes. Consider this diff:

```diff
diff --git a/actionpack/lib/action_dispatch/routing/mapper.rb b/actionpack/lib/action_dispatch/routing/mapper.rb
index 07e3be4db8..c499d90617 100644
--- a/actionpack/lib/action_dispatch/routing/mapper.rb
+++ b/actionpack/lib/action_dispatch/routing/mapper.rb
@@ -390,7 +390,7 @@ def dispatcher(raise_on_name_error)
       # for root cases, where the latter is the correct one.
       def self.normalize_path(path)
         path = Journey::Router::Utils.normalize_path(path)
-        path.gsub!(%r{/(\(+)/?}, '\1/') unless path =~ %r{^/(\(+[^)]+\)){1,2}$}
+        path.gsub!(%r{/(\(+)/?}, '\1/') unless path =~ %r{^/(\(+[^)]+\)){1,}$}
         path
       end
```

File: actionpack/test/journey/router/utils_test.rb [hunk 2]
```diff
@@ -42,6 +42,22 @@ def test_normalize_path_maintains_string_encoding
         def test_normalize_path_with_nil
           assert_equal "/", Utils.normalize_path(nil)
         end
+
```
Comment (gmcgibbon) [inline actionpack/test/journey/router/utils_test.rb]: These tests pass regardless of your change. I think you should add one to `actionpack/test/dispatch/routing_test.rb` instead. Try this:

```ruby
def test_deeply_scoped_root_unscoped_access
  draw do
    scope "(:locale)" do
      scope "(:platform)" do
        scope "(:browser)" do
          root to: "projects#index"
        end
      end
    end
  end

  assert_equal "/", root_path
  get "/"
  assert_equal "projects#index", @response.body
end
```

**Review by gmcgibbon — COMMENTED** [review]
Still an issue on Rails 6 (d8ac08f6bf545c025d7540fb680f41233bcca566). I think you should add support for any number of optional parameter scopes.

**Pull request closed without merging**